File: frontend/src/types/strategy.ts
export interface Strategy {
  id: string;
  name: string;
  description: string | null;
  strategy_type: string;
  parameters: Record<string, unknown>;
  symbols: string[];
  index_filter: string | null;
  timeframe: string;
  is_active: boolean;
  is_paper: boolean;
  created_at: string;
  updated_at: string;
}

export interface Signal {
  id: string;
  strategy_id: string;
  symbol: string;
  signal_type: "buy" | "sell" | "hold";
  confidence: number;
  price: number;
  stop_loss: number | null;
  take_profit: number | null;
  indicators: Record<string, number>;
  created_at: string;
}

export interface TrendCandidate {
  symbol: string;
  name: string;
  price: number;
  score: number;
  trend_status: string;
  rsi: number | null;
  volume_ratio: number | null;
  support_level?: number;
  resistance_level?: number;
  breakout_level?: number;
  target_price?: number;
  macd_signal?: string;
  indicators: Record<string, number>;
}

File: backend/app/schemas/strategy.py
# Source: Doc 03 §4 — Pydantic strategy şemaları
"""Strateji şemaları."""

from datetime import datetime
from decimal import Decimal
from enum import Enum
from uuid import UUID

from pydantic import BaseModel, Field


class SignalType(str, Enum):
    BUY = "buy"
    SELL = "sell"
    HOLD = "hold"


class StrategyCreateRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = None
    strategy_type: str
    parameters: dict = Field(default_factory=dict)
    symbols: list[str] = Field(default_factory=list)
    index_filter: str | None = None
    timeframe: str = "1d"
    risk_params: dict = Field(default_factory=dict)


class StrategyResponse(BaseModel):
    id: UUID
    name: str
    description: str | None = None
    strategy_type: str
    parameters: dict
    symbols: list
    index_filter: str | None = None
    timeframe: str
    is_active: bool
    is_paper: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SignalResponse(BaseModel):
    id: UUID
    strategy_id: UUID
    symbol: str
    signal_type: SignalType
    confidence: Decimal
    price: Decimal
    stop_loss: Decimal | None = None
    take_profit: Decimal | None = None
    indicators: dict
    is_executed: bool
    created_at: datetime

    model_config = {"from_attributes": True}
